src/tests/Microsoft.PowerFx.Interpreter.Tests/CustomFunctions.cs

```diff
@@ -44,6 +44,138 @@ public static StringValue Execute(NumberValue x, BooleanValue b)
             }
         }
 
+        [Fact]
+        public async void CustomFunction_CallBack()
+        {
+            var config = new PowerFxConfig(null);
+            config.AddFunction(new TestCallbackFunction());
+            var engine = new RecalcEngine(config);
+
+            // Shows up in enuemeration
+            var func = engine.GetAllFunctionNames().First(name => name == "TestCallback");
+            Assert.NotNull(func);
+
+            var result = engine.Eval("TestCallback(1=2)");
+            Assert.Equal(false, result.ToObject());
+        }
+
+        private class TestCallbackFunction : ReflectionFunction
+        {
+            // Must have "Execute" method. 
+            // Any arg can be a boolean callback function.
+            public static BooleanValue Execute(Func<Task<BooleanValue>> expression)
+            {
+                return expression().Result;
+            }
+        }
+
+        [Fact]
+        public async void CustomFunctionAsync_CallBack()
+        {
+            var config = new PowerFxConfig(null);
+            config.AddFunction(new WaitAsyncFunction());
+            config.AddFunction(new HelperFunction(x => FormulaValue.New(x.Value + 1)));
+            var engine = new RecalcEngine(config);
+
+            // Shows up in enuemeration
+            var func = engine.GetAllFunctionNames().First(name => name == "WaitAsync");
+            Assert.NotNull(func);
+
+            // Can be invoked. 
+            using var cts = new CancellationTokenSource();
+            var result = engine.EvalAsync("WaitAsync(Helper() = 3)", cts.Token);
```

Comment:
>WaitAsync [](http://example.com/codeflow?start=43&length=9)

nit: Power Fx convention is to not use "Async" suffix since everything is potentially async. 
So just rename WaitAsyncFunction to WaitFunction